### ISSUE TITLE
id sequences

### ISSUE THREAD

**Issue by Ragura** [factory-js/factory-js#68]
### Describe your proposal

Hi there, thanks for your work on this package! I've used factory boy before and I like how the api is sort of similar to work with, but this time with actual type support unlike in Python.

I might have missed it, but this example uses a user id that I don't see defined anywhere on the factory:

```ts
const userFactory = await factory
  .define({
    props: {
      firstName: () => "John",
      lastName: () => "Doe",
    },
    vars: {},
  })
  .after((user) => {
    postFactory.props({ userId: () => user.id }).buildList(3);
  });
 ```

How can we set it up so that ids generated by factories follow a specific sequence? Same goes for other places where a sequence might be interesting to have.

**Comment by ktmouk** [comment]
Hi, Ragura 👋 

Your question might be similar with this [discussion](https://github.com/orgs/factory-js/discussions/28#discussioncomment-9558934). FactoryJS itself does not have a `seq` feature, but you can generate sequence numbers by adding the small function mentioned in the discussion.

```ts
const seq = <O>(iteratee: (count: number) => Promise<O> | O) => {
  let count = 0;
  return () => iteratee(count++);
};

const userFactory = await factory
  .define({
    props: {
      id: seq((n) => n)
    },
    vars: {}
  })

console.log(await userFactory.build()) // { id: 0 }
console.log(await userFactory.build()) // { id: 1 }
console.log(await userFactory.build()) // { id: 2 }
```

If you think FactoryJS should provide this function by default, please let us know your use case.  
Also, the example you mentioned is just my mistake. I'll fix it later.

**Comment by Ragura** [comment]
Thanks for the suggestion, that works! I think it would still have merit to include this functionality in the factory definition options, simply because it could be used in different situations as well. For example, having an ItemFactory with generated names such as `item-1`, `item-2`, ... when using `buildList()`. There's nothing preventing us from duplicating the helper function where needed, but it would just add boilerplate code.

**Comment by ktmouk** [comment]
Thank you for sharing your opinion!

I looked into the Vitest docs and found that Vitest and Jest have a [WORKER_ID env](https://vitest.dev/guide/migration.html#envs), so I realized we can use `seq` and WORKER_ID to avoid the UNIQUE constraint violation like this:

```ts
const userFactory = await factory
  .define({
    props: {
      email: seq((n) => {
        return `${n}-${process.env['VITEST_WORKER_ID']}@example.com`
      }),
    },
    vars: {}
  })
```

I generally recommend using UUID values to avoid UNIQUE constraint violations as it is simpler, but the above solution could be helpful in cases where we need to generate unique numbers, such as a phone number.

I agree that FactoryJS should have `seq` function by default. Are you interested in contributing it?
I think we can reimplement it by just reverting [this PR](https://github.com/factory-js/factory-js/pull/17).
